### DISCUSSION TITLE
Add support for promo codes that change the invoice price by a %

### DISCUSSION THREAD

**Discussion by ai-oleynikov** [btcpayserver/btcpayserver#5425] [Ideas / Feature Requests]
The current implementation of [promo codes](https://docs.btcpayserver.org/Forms/#promo-codes) only supports flat discounts. However, BTCPay Server users expect to be able to offer discounts as a percentage of the invoice price:
- https://t.me/btcpayserver/64221
- https://t.me/btcpayserver/64292

**Comment by aeonBTC** [comment]
Definitely a needed feature. Promo codes that aren't percentage based are essentially useless for any merchant trying to setup referral programs and discounts.

**Reply by dennisreimann** [reply]
This is getting addressed in #5463.
